### ISSUE TITLE
RTIO Improvements

### ISSUE THREAD

**Issue by teburd** [zephyrproject-rtos/zephyr#46658]
- [ ] rtio_sqe_suspend when the needed iodev is busy to inform the executor to suspend the chain
- [ ] add test for rtio_submit with active queue already in progress (should mark the queue correctly but not start the callback chain)
- [ ] RTIO_OP_PIPE_FLUSH io_uring equivalent to wait on all prior submissions
- [ ] Tests and code simplification for suspending of threads on submit or cq consume.
- [ ] DMA usage experiments (policy based, iodev specific, needs some experimentation)
- [ ] Optional submission timeouts
- [ ] Sample with concurrent chains and real devices

**Comment by teburd** [comment]
@yperess added the provided buffer feature, let me know if that makes sense here or if it could use its own ticket
